docs/book/src/topics/workload-identity.md

```diff
@@ -0,0 +1,204 @@
+
+- At this stage, you will need to create a federated identity credential.
+  - You can create that either with Azure AD application or user-assigned
+    identity. Please note that user assigned identity will need to be created
+    regardless because cloud provider azure integration is not yet done. The
+    steps are mentioned in the next section of workload cluster creation.
+  - The next list items links to steps on creating the federated
+    identity credential. You will need to set up several environment
+    variables:
+    - `SERVICE_ACCOUNT_NAMESPACE` : Namespace where the capz-manager pod
+      will run.
+    - `SERVICE_ACCOUNT_NAME` : Name of the capz-manager k8s service account.
+    - `SERVICE_ACCOUNT_ISSUER` : This is the path of the Azure storage
+      container which you created in the previous step which is:
+      `"https://${AZURE_STORAGE_ACCOUNT}.blob.core.windows.net/${AZURE_STORAGE_CONTAINER}/"`
+
+  - Create a federated identity credential using the steps outlined [here](https://azure.github.io/azure-workload-identity/docs/topics/federated-identity-credential.html)
+    You can either use `user-assigned-identity` or `AD application` to create federated identity credential.
+
+- Create a Kind cluster with necessary flags with the following command:
+
+```bash
+cat <<EOF | kind create cluster --name azure-workload-identity --config=-
+kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+nodes:
+- role: control-plane
+  extraMounts:
+    - hostPath: ${SERVICE_ACCOUNT_KEY_FILE}
+      containerPath: /etc/kubernetes/pki/sa.pub
+    - hostPath: ${SERVICE_ACCOUNT_SIGNING_KEY_FILE}
+      containerPath: /etc/kubernetes/pki/sa.key
+  kubeadmConfigPatches:
+  - |
+    kind: ClusterConfiguration
+    apiServer:
+      extraArgs:
+        service-account-issuer: ${SERVICE_ACCOUNT_ISSUER}
+        service-account-key-file: /etc/kubernetes/pki/sa.pub
+        service-account-signing-key-file: /etc/kubernetes/pki/sa.key
+    controllerManager:
+      extraArgs:
+        service-account-private-key-file: /etc/kubernetes/pki/sa.key
+EOF
+```
+
+- Initialize a management cluster using `clusterctl` using the below command.
+  If you do not have `clusterctl` installed, then follow this [link](https://cluster-api.sigs.k8s.io/user/quick-start.html#install-clusterctl)
+  to install.
+```bash
+$ clusterctl init --infrastructure azure
+```
+
+### Creating a Workload Cluster
+
+- Create a user-assigned identity using the below steps:
+  - [Create](https://learn.microsoft.com/en-gb/azure/active-directory/managed-identities-azure-resources/how-manage-user-assigned-managed-identities?pivots=identity-mi-methods-azp#create-a-user-assigned-managed-identity)
+a user-assigned managed identity in Azure.
+  - [Create a role assignment](https://learn.microsoft.com/en-us/azure/role-based-access-control/role-assignments-portal-managed-identity)
```

Comment:
Might be worth adding a note to save `USER_ASSIGNED_IDENTITY_NAME` after creating it.

Reply:
Sure. Will add it